### PR TITLE
Annotation model

### PR DESCRIPTION
**Pull request by southeo**
Update annotation model according to [Annotation FDO Record RFC](https://docs.google.com/document/d/1uOPXZGvRmSWgnOYoDVEr02EAeC3ZV_5wCdjtdTKaR3o/edit)

Did some refactoring of the various enums we have so it looks like a lot of files were affected.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -82,6 +83,11 @@
       <groupId>org.springframework.boot</groupId>
       <artifactId>spring-boot-starter-oauth2-resource-server</artifactId>
     </dependency>
+    <dependency>
```
Comment (samleeflang) [inline pom.xml:86]: Why do we need netty-codec?

File: src/main/java/eu/dissco/core/handlemanager/service/FdoRecordService.java [hunk 2]
```diff
@@ -354,32 +354,23 @@ public List<HandleAttribute> prepareAnnotationAttributes(AnnotationRequest reque
...
+    fdoRecord.add(new HandleAttribute(TARGET_PID, handle, request.getTargetPid()));
 
-    // 502 replaceOrAppend
-    fdoRecord.add(
-        new HandleAttribute(REPLACE_OR_APPEND, handle, request.getReplaceOrAppend().toString()));
+    // 501 TargetType
+    fdoRecord.add(new HandleAttribute(TARGET_TYPE, handle, request.getTargetType()));
 
-    // 503 AccessRestricted
+    // 503 motivation
```
Comment (samleeflang) [inline src/main/java/eu/dissco/core/handlemanager/service/FdoRecordService.java:364]: I guess this needs to be 502?

File: src/test/java/eu/dissco/core/handlemanager/testUtils/TestUtils.java [hunk 3]
```diff
@@ -611,31 +612,25 @@ public static List<HandleAttribute> genMediaObjectAttributes(byte[] handle,
...
-    // 503 AccessRestricted
-    fdoRecord.add(new HandleAttribute(ACCESS_RESTRICTED.index(), handle, ACCESS_RESTRICTED.get(),
-        String.valueOf(false).getBytes(StandardCharsets.UTF_8)));
+    // 501 TargetType
+    fdoRecord.add(new HandleAttribute(TARGET_TYPE, handle, TARGET_TYPE_TESTVAL));
 
-    // 504 LinkedObjectUrl
-    fdoRecord.add(new HandleAttribute(LINKED_OBJECT_URL.index(), handle, LINKED_OBJECT_URL.get(),
-        LINKED_URL_TESTVAL.getBytes(StandardCharsets.UTF_8)));
+    // 503 motivation
```
Comment (samleeflang) [inline src/test/java/eu/dissco/core/handlemanager/testUtils/TestUtils.java:625]: Probably needs to be 502

**Review by samleeflang — COMMENTED** [review]
Just one or two small comments

File: pom.xml (hunk 1 above)
Comment (southeo) [reply pom.xml:86]: To force the correct version for trivy

File: Dockerfile [hunk 4]
```diff
@@ -1,6 +1,5 @@
 FROM eclipse-temurin:17-alpine AS builder
-RUN apk update
-RUN apk upgrade openssl
+RUN apk update --no-cache && apk upgrade openssl
```
Comment (samleeflang) [inline Dockerfile:2]: This didn't help?

File: src/main/java/eu/dissco/core/handlemanager/service/FdoRecordService.java [hunk 5]
```diff
@@ -114,669 +30,662 @@
...
+    public List<HandleAttribute> prepareAnnotationAttributes(AnnotationRequest request, byte[] handle,
+                                                             ObjectType type)
+            throws UnprocessableEntityException, PidResolutionException, InvalidRequestException {
+        var fdoRecord = prepareHandleRecordAttributes(request, handle, type);
 
-    // 500 TargetPid
-    resolveInternalPid(request);
-    fdoRecord.add(new HandleAttribute(TARGET_PID, handle, request.getTargetPid()));
+        // 500 TargetPid
+        //resolveInternalPid(request); -> Do we want to validate target PID?
```
Comment (samleeflang) [inline src/main/java/eu/dissco/core/handlemanager/service/FdoRecordService.java:280]: This is not a question for in the code! This is a question for a refinement.... sneaking in code comment....

File: Dockerfile (hunk 4 above)
Comment (southeo) [reply Dockerfile:2]: It didn't, as far as I can tell.

**Review by samleeflang — APPROVED** [review]
:popcorn:

**Pull request merged**